bakery/src/scripts/fetch_map_resources.py

```diff
@@ -1,35 +1,127 @@
 """Map resource files used in CNXML to provided path"""
 
+import hashlib
+import json
+import magic
+import shutil
 import sys
 from pathlib import Path
 from lxml import etree
 
+# same as boto3 default chunk size. Don't modify.
+BUF_SIZE = 8 * 1024 * 1024
+
+# relative links must work both locally, on PDF, and on REX, and images are
+# uploaded with the prefix 'resources/' in S3 for REX
+# so the output directory name MUST be resources
+RESOURCES_DIR_NAME = 'resources'
+
+
+def create_json_metadata(output_dir, sha1, mime_type, s3_md5, original_name):
+    """ Create json with MIME type of a (symlinked) resource file """
+    data = {}
+    data['original_name'] = original_name
+    data['mime_type'] = mime_type
+    data['s3_md5'] = s3_md5
+    data['sha1'] = sha1
+    json_file = output_dir / f'{sha1}.json'
+    with json_file.open(mode='w') as outfile:
+        json.dump(data, outfile)
+
+
+def get_mime_type(filename):
+    """ get MIME type of file with libmagic """
+    mime_type = ''
+    try:
+        mime_type = magic.from_file(filename, mime=True)
+    finally:
+        return mime_type
+
+
+# https://stackoverflow.com/a/22058673/756056
+def get_checksums(filename):
```

Comment:
It looks like `get_checksums` and `get_mime_type` are the same (`create_json_metadata` is slightly modified) as `checksum_resource.py`. Can we import some of these from there or move to `utils.py` to share the implementation source?